src/main/java/graphql/language/DirectivesContainer.java

```diff
@@ -6,34 +6,67 @@
 import java.util.List;
 import java.util.Map;
 
-import static graphql.language.NodeUtil.directivesByName;
+import static graphql.language.NodeUtil.allDirectivesByName;
+import static graphql.language.NodeUtil.nonRepeatableDirectivesByName;
+import static graphql.language.NodeUtil.nonRepeatedDirectiveByNameWithAssert;
+import static java.util.Collections.emptyList;
 
 /**
- * Represents a language node that can contain Directives.
+ * Represents a language node that can contain Directives.  Directives can be repeatable and (by default) non repeatable.
+ * <p>
+ * There are access methods here that get the two different types.
+ *
+ * @see graphql.language.DirectiveDefinition
+ * @see DirectiveDefinition#isRepeatable()
  */
 @PublicApi
 public interface DirectivesContainer<T extends DirectivesContainer> extends Node<T> {
```

Comment:
This method pattern is repeated in GraphqlDirectiveContainer as well - same names - same pattern - different types